infra/core/networking/vnet.bicep

```diff
@@ -0,0 +1,73 @@
+param location string
+param name string
+param tags object = {}
+param appServicePlanName string
+param appServicePlanId string
+
+var addressPrefix = '10.0.0.0/16'
+
+var subnets = [
+  {
+    name: 'ai-subnet'
+    properties: {
+      addressPrefix:'10.0.1.0/24'
+      privateEndpointNetworkPolicies: 'Enabled'
+      privateLinkServiceNetworkPolicies: 'Enabled'      
+    }
+  }
+  {
+    name: 'AzureBastionSubnet'
+    properties: {
+      addressPrefix:'10.0.2.0/24'
+      privateEndpointNetworkPolicies: 'Enabled'
+      privateLinkServiceNetworkPolicies: 'Enabled'      
+    }
+  }
+  {
+    name: 'app-int-subnet'
+    properties: {
+      addressPrefix:'10.0.3.0/24'
+      privateEndpointNetworkPolicies: 'Enabled'
+      privateLinkServiceNetworkPolicies: 'Enabled'
+      delegations: [
+        {
+          id: appServicePlanId
+          name: appServicePlanName
+          properties: {
+            serviceName: 'Microsoft.Web/serverFarms'
+          }
+        }
+      ]   
+    }
+  }  
+]
+
+resource vnet 'Microsoft.Network/virtualNetworks@2021-02-01' = {
+  name: name
+  location: location
+  tags: tags
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        addressPrefix
+      ]
+    }
+    subnets: subnets
+  }
+}
+
+output subnets array = [for (name, i) in subnets :{
+  subnets : vnet.properties.subnets[i]
+}]
+
+output subnetids array = [for (name, i) in subnets :{
+  subnets : vnet.properties.subnets[i].id
+}]
+
+
+output id string = vnet.id
+output name string = vnet.name
+
+output aiSubId string = vnet.properties.subnets[0].id
+output bastionSubId string = vnet.properties.subnets[1].id
+output appIntSubId string = vnet.properties.subnets[2].id
```

Comment:
We can probably have longer names here so clear what ai means

Reply:
I'm confused myself as to what "ai" means. Is it meant to mean AI? We probably want something more generic here since this is in core. @tonybaloney ?

Reply:
I think the subnets should be a param to the file instead, I'll refactor it.

Reply:
And the `ai` subnet is all backend services

Reply:
See https://github.com/Azure-Samples/azure-search-openai-demo/pull/864/commits/d5ff641a5ed610159dd8ff84423c9fa9cc9926ab for changes